Lib/test/test_typing.py

```diff
@@ -8052,6 +8101,23 @@ class T4(TypedDict, Generic[S]): pass
                 self.assertEqual(klass.__optional_keys__, set())
                 self.assertIsInstance(klass(), dict)
 
+    def test_setname_called_on_things_in_class_namespace(self):
+        class CustomException(Exception): pass
+
+        class Annoying:
+            def __set_name__(self, owner, name):
+                raise CustomException("Cannot do that!")
+
+        with self.assertRaisesRegex(CustomException, "Cannot do that!") as cm:
+            class Foo(TypedDict):
+                attr = Annoying()
```

Comment:
Not sure that TypeDict classes allow any class-level attributes 🤔 

https://mypy-play.net/?mypy=latest&python=3.11&gist=5a7a905e4a1624956eff5c9aa6e33365

> The class body should only contain lines with item definitions of the form key: value_type, optionally preceded by a docstring. The syntax for item definitions is identical to attribute annotations, but there must be no initializer, and the key name actually refers to the string value of the key instead of an attribute name.

Reply:
Yeah. Do you think I should remove this test, then?

On the one hand, it felt useful to assert that TypedDict class namespaces called `__set_name__` in the same way as any other class namespace. On the other hand, as you say, TypedDicts are very special, and you're not really _meant_ to have any class-level attributes in a TypedDict "class" — so maybe we don't care

Reply:
I think that we should remove the test, yes.

Reply:
Done!